wagtail/admin/views/pages/bulk_actions/move.py

```diff
@@ -0,0 +1,100 @@
+from django import forms
+from django.shortcuts import get_object_or_404
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext
+
+from wagtail.admin import widgets
+from wagtail.admin.views.pages.bulk_actions.page_bulk_action import PageBulkAction
+from wagtail.core import hooks
+from wagtail.core.models import Page
+
+
+class MoveForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        destination = kwargs.pop('destination')
+        super().__init__(*args, **kwargs)
+        self.fields['chooser'] = forms.ModelChoiceField(
+            initial=destination,
+            queryset=Page.objects.all(),
+            widget=widgets.AdminPageChooser(can_choose_root=True, user_perms='move_to'),
+            label=_("Select a new parent page"),
+        )
+
+
+def before_bulk_move(request, action_type, pages, action_instance):
```

Comment:
I don't think this should be done via a hook, because we're writing the original view in the first place.